cmd/frontend/graphqlbackend/executor_secrets.go

```diff
@@ -0,0 +1,272 @@
+package graphqlbackend
+
+import (
+	"context"
+	"regexp"
+	"strings"
+
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/graphqlutil"
+	"github.com/sourcegraph/sourcegraph/internal/actor"
+	"github.com/sourcegraph/sourcegraph/internal/auth"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/encryption/keyring"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+var executorSecretKeyPattern = regexp.MustCompile("^[A-Z][A-Z|0-9|_]*$")
```

Comment:
This is currently allowing `|` characters as well, so:

```suggestion
var executorSecretKeyPattern = regexp.MustCompile("^[A-Z][A-Z0-9_]*$")
```

Reply:
Beyond that, this _may_ be a little too restrictive — lower case letters are allowed in environment variables, and may even be useful in some cases such as `http_proxy` — but we could probably wait and see if we get any complaints before changing this.

Reply:
Duh, thanks for catching that! 

Yeah I am not sure about this either. 

> Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001 consist solely of uppercase letters, digits, and the '_' (underscore) from the characters defined in Portable Character Set and do not begin with a digit. Other characters may be permitted by an implementation; applications shall tolerate the presence of such names.

I kinda followed this guide, but it might be too restrictive in the end. Since we pass the env vars to the shell, it might be that some exotic ones wouldn't work though. 
Happy to make it more lax, if you have any suggestion what would be good to cover as well, for now I was mostly scared that people put in invalid values and that breaks execution 😬